README.md

```diff
@@ -207,23 +220,273 @@ More information can be found [here](http://www.diveintopython3.net/strings.html
+```r
+toOmit <- c("copyFileHandles", "notifyMe")
+selectFunctions <- function(x) {
+    if (any(x$name == toOmit)) {
+        return(NULL)
+    }
+    x
+}
+```
+
+Then from the R script that generates .Rd files, update `generateRdFiles` as following:
+```r
+generateRdFiles(srcRootDir,
+                pyPkg = "synapseutils",
+                module = "synapseutils",
+                modifyFunctions = selectFunctions)
+```
+
+And in your `.onLoad`, update `generateRWrappers` as following:
+```r
+generateRWrappers(pyPkg = "synapseutils",
+                  module = "synapseutils",
+                  setGenericCallback = callback,
+                  modifyFunctions = selectFunctions)
+```
+
+### Expose a subset of classes within a Python module
+
+The `synapseclient` module in `synapsePythonClient` has the following structure:
+```
+    module: synapseclient
+        module: client
+            class: Synapse
+                function: get
+                function: login
+        module: entity
+            class: Entity
+                function: get
+                function: privateGet
+            class: File
+                function: get
+                function: privateGet
+            class: Folder
+                function: get
+                function: privateGet
+```
+
+Now let's try a more complicated example where we want to expose the following functions and classes from `synapseclient.entity` module:
+```
+  class: File
+      function: get
+  class: Folder
+      function: get
+```
+
+Note that we do not want the following from the `synapseclient.entity` module:
+* `function: privateGet` from any of the classes
+* `class: Entity`
+
+First, define your `selectClasses` in the shared .R file:
+```r
+methodsToOmit <- "privateGet"
+classToSkip <- "Entity"
+selectClasses <- function(class) {
+    if (any(class$name == classToSkip)) {
+        return(NULL)
+    }
+    if (!is.null(class$methods)) {
+        culledMethods <- lapply(X = class$methods, function(x) {
+            if (any(x$name == methodsToOmit)) NULL else x;
+        })
+        # Now remove the nulls
+        nullIndices <- sapply(culledMethods, is.null)
+        if (any(nullIndices)) {
+            class$methods <- culledMethods[-which(nullIndices)]
+        }
+    }
+    class
+}
+```
+
+Then call `generateRdFiles` and `generateRWrappers` as following:
+```r
+generateRdFiles(srcRootDir,
+                pyPkg = "synapseclient",
+                module = "synapseclient.entity",
+                modifyClasses = selectClasses)
+```
+```r
+generateRWrappers(pyPkg = "synapseclient",
+                  module = "synapseclient.entity",
+                  setGenericCallback = callback,
+                  modifyClasses = selectClasses)
+```
+
+### Expose functions within a singleton object
+
+In rare cases, we want to expose Python functions under `synapseclient.client.Synapse` without exposing the `Synapse` object itself.
```

Comment:
In rare cases, we want to expose a set Python functions which are an object's methods, but without exposing the object itself.  For example, in the `synapser` package we wish to make available the methods in `synapseclient.client.Synapse` without requiring the R user to instantiate the `Synapse` object.  The following will create a singleton object at package load time and expose the object's methods to be called directly.